core/store/orm/orm.go

```diff
@@ -337,10 +337,13 @@ func (orm *ORM) SaveJobRun(run *models.JobRun) error {
 			Where("updated_at = ?", run.UpdatedAt).
 			Omit("deleted_at").
 			Save(run)
+		if result.Error != nil {
```

Comment:
It looks like this could have treated say a connection error mistakenly as an `OptimisticUpdateConflictError`, could we add a test?

Reply:
I did add a test ;)

Reply:
I don't see it.

Reply:
https://github.com/smartcontractkit/chainlink/pull/2629/files#diff-63221a8015f37683ea3beacc6001efe9R166